### PR TITLE
Clean up infrastructure and API

### PR DESCRIPTION
**Pull request by astrofrog**
This cleans up the infrastructure (to use e.g. setuptools_scm and tox) and also makes it so that the public API is at the top level. There are currently four functions in the public API (see docs preview).

One question is whether we want to explicitly rename the sub-packages to include an underscore to make sure people switch to using the top level import when migrating to extension-helpers?

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    py{36,37,38}-test
+    build_docs
+    pycodestyle
+requires =
+    setuptools >= 30.3.0
+    pip >= 19.3.1
+isolated_build = true
+
+[testenv]
+passenv =
```
Comment (Cadair) [inline tox.ini]: Do we need all of these here?

Comment (astrofrog) [reply tox.ini]: No, I'll clean this up

File: azure-pipelines.yml [hunk 2]
```diff
@@ -1 +1,34 @@
-#
+variables:
+  OPENMP_EXPECTED: True
```
Comment (Cadair) [inline azure-pipelines.yml]: remove this?

File: extension_helpers/tests/test_openmp_helpers.py [hunk 3]
```diff
@@ -5,31 +5,37 @@
...
+@pytest.fixture
+def openmp_expected(request):
+    openmp_expected = request.config.getoption("--openmp-expected")
+    if openmp_expected is not None:
+        return openmp_expected.lower() == 'true'
+
+
+def test_add_openmp_flags_if_available(openmp_expected):
+
+    print(openmp_expected)
```
Comment (Cadair) [inline extension_helpers/tests/test_openmp_helpers.py]: remove print and extra blank lines.

File: extension_helpers/tests/test_openmp_helpers.py [hunk 4]
```diff
@@ -5,31 +5,37 @@
...
-    else:
-        assert not using_openmp
+    if openmp_expected is not None:
+        assert openmp_expected is using_openmp
+
 
+def test_generate_openmp_enabled_py(openmp_expected):
 
-def test_generate_openmp_enabled_py():
+    print(openmp_expected)
```
Comment (Cadair) [inline extension_helpers/tests/test_openmp_helpers.py]: remove print.

File: extension_helpers/tests/test_openmp_helpers.py (hunk 3 above)
Comment (astrofrog) [reply extension_helpers/tests/test_openmp_helpers.py]: Done

File: azure-pipelines.yml (hunk 2 above)
Comment (astrofrog) [reply azure-pipelines.yml]: Done

File: extension_helpers/tests/test_openmp_helpers.py (hunk 4 above)
Comment (astrofrog) [reply extension_helpers/tests/test_openmp_helpers.py]: Done

**Pull request merged**